File: Core/TouchToolkit/CS/TouchToolkitExplorer/GcZoomPages/GcZoomCommenDemo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using TouchToolkitExplorer.C1ZoomPages.Help_Forms;
using C1.Win.TouchToolKit;

namespace TouchToolkitExplorer.C1ZoomPages
{
    public partial class C1ZoomCommenDemo : DemoBase
    {
        public C1ZoomCommenDemo()
        {
            InitializeComponent();
            InitializeDemo();
        }

        C1Zoom _gcZoom = new C1Zoom();

        private void InitializeDemo()
        {
            this.Title = "C1Zoom Basic (Zoom and Scroll)";
            this.Description = @"This page is used to try basic features of C1Zoom. 
  - Click the Show Form Button to show a test form. And test the Gestrue zoom/scroll.
  - Try to change the settings to knows how to config the C1Zoom.";

            AddEnumItemsToComboBox(ZoomPreviewModeComboBox, typeof(ZoomPreviewMode));
            AddEnumItemsToComboBox(scrollIndicatorCombo, typeof(ScrollIndicatorMode));
            formatComboBox.Items.Add("Zooming {Percentage}%");
            formatComboBox.Items.Add("Zooming x{ZoomFactor}");

            scrollIndicatorCombo.SelectedItem = _gcZoom.ScrollIndicatorMode;
            ZoomPreviewModeComboBox.SelectedItem = _gcZoom.ZoomPreviewMode;
            formatComboBox.SelectedIndex = 0;
            displayLabel.Font = _gcZoom.AlternativeContentSettings.Font;
            displayLabel.ForeColor = _gcZoom.AlternativeContentSettings.ForeColor;
            HorizontalRailEnableCheckBox.DataBindings.Add("Checked", _gcZoom, "IsHorizontalRailEnabled");
            VerticalRailEnableCheckBox.DataBindings.Add("Checked", _gcZoom, "IsVerticalRailEnabled");
            AllowDoubleTapZoomCheckBox.DataBindings.Add("Checked", _gcZoom, "AllowDoubleTapZoom");
            KeepAspectRatioCheckBox.DataBindings.Add("Checked", _gcZoom, "KeepAspectRatio");
            scrollIndicatorCombo.DataBindings.Add("SelectedItem", _gcZoom, "ScrollIndicatorMode");
            boundaryFeedbackUI1.DataBindings.Add("BoundaryFeedbackMode", _gcZoom, "BoundaryFeedbackMode");

            ZoomPreviewModeComboBox.SelectedIndexChanged += ZoomPreviewModeComboBox_SelectedIndexChanged;
            formatComboBox.TextChanged += formatComboBox_TextChanged;
            FontButton.Click += FontButton_Click;
            ForeColorButton.Click += ForeColorButton_Click;
            formButton.Click += formButton_Click;
            boundaryFeedbackUI1.BoundaryFeedbackModeChanged += boundaryFeedbackUI1_BoundaryFeedbackModeChanged;
        }

        void formButton_Click(object sender, EventArgs e)
        {
            using (C1ZoomCommenForm form = new C1ZoomCommenForm())
            {
                _gcZoom.Target = form;
                form.ShowDialog(this);
                // The _gcZoom will auto detach when the Form dispose.
                //_gcZoom.SetEnabled(form, false);
            }
        }

        void ZoomPreviewModeComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            _gcZoom.ZoomPreviewMode = (ZoomPreviewMode)ZoomPreviewModeComboBox.SelectedItem;

            this.alternativeContentGroup.Visible = false;
            this.NoPreivewInfoPanel.Visible = false;
            this.boundaryFeedbackUI1.Visible = false;
            switch (_gcZoom.ZoomPreviewMode)
            {
                case ZoomPreviewMode.Bitmap:
                    this.boundaryFeedbackUI1.Visible = true;
                    break;
                case ZoomPreviewMode.NoPreview:
                    this.NoPreivewInfoPanel.Visible = true;
                    break;
                case ZoomPreviewMode.AlternativeContent:
                    this.alternativeContentGroup.Visible = true;
                    break;
                default:
                    break;
            }
        }

        void formatComboBox_TextChanged(object sender, EventArgs e)
        {
            if (formatComboBox.Text != null)
            {
                _gcZoom.AlternativeContentSettings.Format = formatComboBox.Text;
                displayLabel.Text = formatComboBox.Text.Replace("{Percentage}", "100").Replace("{ZoomFactor}", "1.00"); ;
            }
        }

        void ForeColorButton_Click(object sender, EventArgs e)
        {
            ColorDialog dialog = new ColorDialog();
            if (DialogResult.OK == dialog.ShowDialog())
            {
                _gcZoom.AlternativeContentSettings.ForeColor = dialog.Color;
                displayLabel.ForeColor = dialog.Color;
            }
        }

        void FontButton_Click(object sender, EventArgs e)
        {
            FontDialog dialog = new FontDialog();
            if (DialogResult.OK == dialog.ShowDialog())
            {
                _gcZoom.AlternativeContentSettings.Font = dialog.Font;
                displayLabel.Font = dialog.Font;
            }
        }

        void boundaryFeedbackUI1_BoundaryFeedbackModeChanged(object sender, EventArgs e)
        {
            _gcZoom.BoundaryFeedbackMode = boundaryFeedbackUI1.BoundaryFeedbackMode;
        }

        private void AddEnumItemsToComboBox(ComboBox comboBox, Type enumType)
        {
            Array enumArray = Enum.GetValues(enumType);
            for (int i = 0; i < enumArray.Length; i++)
            {
                comboBox.Items.Add(enumArray.GetValue(i));
            }
        }
    }
}

File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcZoomPages/GcZoomCommenDemo.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit

Imports TouchToolKitDemoVB.C1ZoomPages.Help_Forms

Namespace C1ZoomPages
	Partial Public Class C1ZoomCommenDemo
		Inherits DemoBase
		Public Sub New()
			InitializeComponent()
			InitializeDemo()
		End Sub

		Private _gcZoom As New C1Zoom()

		Private Sub InitializeDemo()
			Me.Title = "C1Zoom Basic (Zoom and Scroll)"
			Me.Description = "This page is used to try basic features of C1Zoom. " & vbCr & vbLf & "  - Click the Show Form Button to show a test form. And test the Gestrue zoom/scroll." & vbCr & vbLf & "  - Try to change the settings to knows how to config the C1Zoom."

			AddEnumItemsToComboBox(ZoomPreviewModeComboBox, GetType(ZoomPreviewMode))
			AddEnumItemsToComboBox(scrollIndicatorCombo, GetType(ScrollIndicatorMode))
			formatComboBox.Items.Add("Zooming {Percentage}%")
			formatComboBox.Items.Add("Zooming x{ZoomFactor}")

			scrollIndicatorCombo.SelectedItem = _gcZoom.ScrollIndicatorMode
			ZoomPreviewModeComboBox.SelectedItem = _gcZoom.ZoomPreviewMode
			formatComboBox.SelectedIndex = 0
			displayLabel.Font = _gcZoom.AlternativeContentSettings.Font
			displayLabel.ForeColor = _gcZoom.AlternativeContentSettings.ForeColor
			HorizontalRailEnableCheckBox.DataBindings.Add("Checked", _gcZoom, "IsHorizontalRailEnabled")
			VerticalRailEnableCheckBox.DataBindings.Add("Checked", _gcZoom, "IsVerticalRailEnabled")
			AllowDoubleTapZoomCheckBox.DataBindings.Add("Checked", _gcZoom, "AllowDoubleTapZoom")
			KeepAspectRatioCheckBox.DataBindings.Add("Checked", _gcZoom, "KeepAspectRatio")
			scrollIndicatorCombo.DataBindings.Add("SelectedItem", _gcZoom, "ScrollIndicatorMode")
			boundaryFeedbackUI1.DataBindings.Add("BoundaryFeedbackMode", _gcZoom, "BoundaryFeedbackMode")

			AddHandler ZoomPreviewModeComboBox.SelectedIndexChanged, AddressOf ZoomPreviewModeComboBox_SelectedIndexChanged
			AddHandler formatComboBox.TextChanged, AddressOf formatComboBox_TextChanged
			AddHandler FontButton.Click, AddressOf FontButton_Click
			AddHandler ForeColorButton.Click, AddressOf ForeColorButton_Click
			AddHandler formButton.Click, AddressOf formButton_Click
			AddHandler boundaryFeedbackUI1.BoundaryFeedbackModeChanged, AddressOf boundaryFeedbackUI1_BoundaryFeedbackModeChanged
		End Sub

		Private Sub formButton_Click(sender As Object, e As EventArgs)
			Using form As New C1ZoomCommenForm()
				_gcZoom.Target = form
				' The _gcZoom will auto detach when the Form dispose.
				'_gcZoom.SetEnabled(form, false);
				form.ShowDialog(Me)
			End Using
		End Sub

		Private Sub ZoomPreviewModeComboBox_SelectedIndexChanged(sender As Object, e As EventArgs)
			_gcZoom.ZoomPreviewMode = DirectCast(ZoomPreviewModeComboBox.SelectedItem, ZoomPreviewMode)

			Me.alternativeContentGroup.Visible = False
			Me.NoPreivewInfoPanel.Visible = False
			Me.boundaryFeedbackUI1.Visible = False
			Select Case _gcZoom.ZoomPreviewMode
				Case ZoomPreviewMode.Bitmap
					Me.boundaryFeedbackUI1.Visible = True
					Exit Select
				Case ZoomPreviewMode.NoPreview
					Me.NoPreivewInfoPanel.Visible = True
					Exit Select
				Case ZoomPreviewMode.AlternativeContent
					Me.alternativeContentGroup.Visible = True
					Exit Select
				Case Else
					Exit Select
			End Select
		End Sub

		Private Sub formatComboBox_TextChanged(sender As Object, e As EventArgs)
			If formatComboBox.Text IsNot Nothing Then
				_gcZoom.AlternativeContentSettings.Format = formatComboBox.Text
				displayLabel.Text = formatComboBox.Text.Replace("{Percentage}", "100").Replace("{ZoomFactor}", "1.00")


			End If
		End Sub

		Private Sub ForeColorButton_Click(sender As Object, e As EventArgs)
			Dim dialog As New ColorDialog()
			If DialogResult.OK = dialog.ShowDialog() Then
				_gcZoom.AlternativeContentSettings.ForeColor = dialog.Color
				displayLabel.ForeColor = dialog.Color
			End If
		End Sub

		Private Sub FontButton_Click(sender As Object, e As EventArgs)
			Dim dialog As New FontDialog()
			If DialogResult.OK = dialog.ShowDialog() Then
				_gcZoom.AlternativeContentSettings.Font = dialog.Font
				displayLabel.Font = dialog.Font
			End If
		End Sub

		Private Sub boundaryFeedbackUI1_BoundaryFeedbackModeChanged(sender As Object, e As EventArgs)
			_gcZoom.BoundaryFeedbackMode = boundaryFeedbackUI1.BoundaryFeedbackMode
		End Sub

		Private Sub AddEnumItemsToComboBox(comboBox As ComboBox, enumType As Type)
			Dim enumArray As Array = [Enum].GetValues(enumType)
			For i As Integer = 0 To enumArray.Length - 1
				comboBox.Items.Add(enumArray.GetValue(i))
			Next
		End Sub
	End Class
End Namespace
